test/kubernetes/e2e/features/virtualhost_options/suite.go

```diff
@@ -0,0 +1,161 @@
+		s.NoError(err, "can delete targetRefManifest")
+		err = s.testInstallation.Actions.Kubectl().DeleteFile(s.ctx, extraVhOManifest)
+		s.NoError(err, "can delete extraVhOManifest")
+		err = s.testInstallation.Actions.Kubectl().DeleteFile(s.ctx, sectionNameVhOManifest)
+		s.NoError(err, "can delete sectionNameVhOManifest")
+		s.testInstallation.Assertions.EventuallyObjectsNotExist(s.ctx, proxyService, proxyDeployment)
+	})
+
+	err := s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, targetRefManifest)
+	s.NoError(err, "can apply targetRefManifest")
+	err = s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, extraVhOManifest)
+	s.NoError(err, "can apply extraVhOManifest")
+	err = s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, sectionNameVhOManifest)
+	s.NoError(err, "can apply sectionNameVhOManifest")
+
+	// Check resources are created for Gateway
+	s.testInstallation.Assertions.EventuallyObjectsExist(s.ctx, proxyService, proxyDeployment)
+
+	// Check header manipulation is applied
+	s.testInstallation.Assertions.AssertEventualCurlResponse(
+		s.ctx,
+		curlPodExecOpt,
+		[]curl.Option{
+			curl.WithHost(kubeutils.ServiceFQDN(proxyService.ObjectMeta)),
+			curl.WithHostHeader("example.com"),
+		},
+		expectedResponseWithFooHeader)
+
+	// Check status is accepted on VirtualHostOption with section name
+	s.testInstallation.Assertions.EventuallyResourceStatusMatchesState(
+		s.getterForMeta(&sectionNameVirtualHostOptionMeta),
+		core.Status_Accepted,
+		"gloo-kube-gateway",
+	)
+	// Check status is warning on VirtualHostOptions not selected for attachment
+	s.testInstallation.Assertions.EventuallyResourceStatusMatchesWarningReasons(
+		s.getterForMeta(&virtualHostOptionMeta),
+		[]string{"conflict with more-specific or older VirtualHostOption"},
+		"gloo-kube-gateway",
+	)
+	s.testInstallation.Assertions.EventuallyResourceStatusMatchesWarningReasons(
+		s.getterForMeta(&extraVirtualHostOptionMeta),
+		[]string{"conflict with more-specific or older VirtualHostOption"},
+		"gloo-kube-gateway",
+	)
+}
+func (s *testingSuite) TestMultipleVirtualHostOptions() {
+	s.T().Cleanup(func() {
+		err := s.testInstallation.Actions.Kubectl().DeleteFile(s.ctx, targetRefManifest)
+		s.NoError(err, "can delete targetRefManifest")
+		err = s.testInstallation.Actions.Kubectl().DeleteFile(s.ctx, extraVhOManifest)
+		s.NoError(err, "can delete extraVhOManifest")
+		s.testInstallation.Assertions.EventuallyObjectsNotExist(s.ctx, proxyService, proxyDeployment)
+	})
+
+	err := s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, targetRefManifest)
+	s.NoError(err, "can apply targetRefManifest")
+	err = s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, extraVhOManifest)
+	s.NoError(err, "can apply extraVhOManifest")
+
+	// Check resources are created for Gateway
+	s.testInstallation.Assertions.EventuallyObjectsExist(s.ctx, proxyService, proxyDeployment)
+
+	// Check header manipulation is applied
```

Comment:
[nit] I think the comment here is off since it's testing `expectedResponseWithoutContentLength` 
```suggestion
	// Check healthy response with no content-length header
```